pkg/yang/entry.go

```diff
@@ -660,6 +676,115 @@ func ToEntry(n Node) (e *Entry) {
 		// when the group is used in multiple locations and the
 		// grouping has a leafref that references outside the group.
 		e = ToEntry(g).dup()
+
+		for _, refine := range s.Refine {
```

Comment:
Given that this is a relatively long block -- can we split it out into its own function that we can also test individually?